docs/conceptual/core/storage.md

```diff
@@ -2,8 +2,54 @@
 title: Storage
 ---
 
-* Key Value Database, RocksDB
+The Substrate uses a simple a key-value data store implemented as a database-backed modified Merkle tree.
 
-* Trie abstraction
+## Key-Value Database
 
-* Runtime Storage abstraction
+Substrate implements it's storage database with [RocksDB](https://rocksdb.org/), a persistent key-value store for fast storage environments.
+
+This is used for all the components of Substrate which require persistent storage such as:
+
+* Substrate clients
+* Substrate light-clients
+* Off-chain workers
+
+## Trie Abstraction
+
+One advantage of using a simple key-value store is that you are able to easily abstract other storage structures on top.
```

Comment:
Can we put a bit more content on _how a Trie is abstracting on top of a K-V store_? I have worked with storage from highest level (SRML), to the lowest (raw RPC to to encoded key) and from the looks of it, honestly, I never really felt like I am working with a Trie here. It always looked more like a bare bone KV store to me.

Reply:
Would need to know what that information would look like...

Implementation details like this could be above the level of "conceptual" docs, but I agree we would want this information in the reference docs.

Reply:
I would say it is interesting to keep in mind as a runtime developper that the trie is way more costy than a standard k-v store.
General idea being that you should rather store a serialized struct with two field at a storage location rather than both field at different location, except if you want to reduce the size of the proof.
That is something that can get quite clear when you apply storage cost (if you have a  base cost and variable cost for size to store).